app/src/main/java/com/github/factotum_sdp/factotum/models/Location.kt

```diff
@@ -84,14 +97,9 @@ class Location(query: String, context: Context) {
         private fun tiramisuResultHandler(query: String, geocoder: Geocoder): List<Address>? {
```

Comment:
It's up to you but I think it would be good for these functions to be able to timeout and return with a failure maybe or that it will try again later. Otherwise we might get blocked in the program. Especially with latches.